pkg/operator/controller/canary/controller_test.go

```diff
@@ -145,3 +147,76 @@ func TestCycleServicePort(t *testing.T) {
 		}
 	}
 }
+
+func Test_deduplicateErrorStrings(t *testing.T) {
+	testCases := []struct {
+		Name           string
+		ErrorMessages  []error
+		ExpectedResult []string
+	}{
+		{
+			Name:           "Empty input",
+			ErrorMessages:  []error{},
+			ExpectedResult: []string{},
+		},
+		{
+			Name: "Multiple errors, no repetition",
+			ErrorMessages: []error{
+				fmt.Errorf("foo"),
+				fmt.Errorf("bar"),
+				fmt.Errorf("baz"),
+				fmt.Errorf("quux"),
+			},
+			ExpectedResult: []string{
+				"foo",
+				"bar",
+				"baz",
+				"quux",
+			},
+		},
+		{
+			Name: "All identical errors",
+			ErrorMessages: []error{
+				fmt.Errorf("foo"),
+				fmt.Errorf("foo"),
+				fmt.Errorf("foo"),
+				fmt.Errorf("foo"),
+			},
+			ExpectedResult: []string{
+				"[Seen 4 times] foo",
+			},
+		},
+		{
+			Name: "Multiple errors, with repetition",
+			ErrorMessages: []error{
+				fmt.Errorf("foo"),
+				fmt.Errorf("bar"),
+				fmt.Errorf("foo"),
+				fmt.Errorf("bar"),
+				fmt.Errorf("baz"),
+				fmt.Errorf("foo"),
+				fmt.Errorf("foo"),
+				fmt.Errorf("quux"),
+				fmt.Errorf("quux"),
+				fmt.Errorf("foo"),
+				fmt.Errorf("foo"),
+				fmt.Errorf("quux"),
+			},
+			ExpectedResult: []string{
+				"[Seen 2 times] bar",
+				"baz",
+				"[Seen 6 times] foo",
+				"[Seen 3 times] quux",
+			},
+		},
```

Comment:
How about adding a test case like the one you showed me?
```
foo
bar
foo
baz
```
which should result into
```
bar
[Seen 2x] foo
baz
```
and not into:
```
[Seen 2x] foo
bar
baz
```